spec/rails_helper.rb

```diff
@@ -47,13 +48,18 @@
   #
   # The different available types are documented in the features, such as in
   # https://relishapp.com/rspec/rspec-rails/docs
+  config.include Devise::TestHelpers, type: :controller
   config.infer_spec_type_from_file_location!
   config.include FactoryGirl::Syntax::Methods
   config.before(:suite) do
-    DatabaseCleaner.strategy = :transaction
     DatabaseCleaner.clean_with(:truncation)
   end
   config.before(:each) do
+    if Capybara.current_driver == :rack_test
```

Comment:
I think this solution is quite nice. When the driver is rack_test 
DatabaseCleaner.clean_with(:transaction)
is used, but when another driver is used (which means js is involved)
DatabaseCleaner.strategy = :truncation
is used.
The motivation for this separation is that :transaction is faster but does not work when js is true